src/julia.h

```diff
@@ -1890,6 +1892,8 @@ typedef struct _jl_task_t {
 // hidden state:
     // id of owning thread - does not need to be defined until the task runs
     _Atomic(int16_t) tid;
+    // threadpool id
+    int8_t threadpoolid;
```

Comment:
`gcstack` currently doesn't need alignment padding so this will add 8 bytes.

Reply:
Turns out that I can't move `priority` to hidden state easily to steal bits for this. I'll look at ways to compress the entire structure in a subsequent PR.